### PR TITLE
Plat 567 test delegation token

### PR DESCRIPTION
**Pull request by MikhailZhukov**
**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: ledger-core/virtual/integration/delegationtoken_test.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+		{
+			name:         "Fail with wrong callee in token",
+			testRailID:   "C5198",
+			fakeCaller:   false,
+			fakeCallee:   true,
+			fakeOutgoing: false,
+		},
+		{
+			name:         "Fail with wrong outgoing in token",
+			testRailID:   "C5198",
```
Comment (ElSacramento) [inline ledger-core/virtual/integration/delegationtoken_test.go:54]: duplicate testRailID

File: ledger-core/virtual/integration/delegationtoken_test.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+			fakeCaller:   false,
+			fakeCallee:   false,
+			fakeOutgoing: true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			t.Log("C5197")
```
Comment (ElSacramento) [inline ledger-core/virtual/integration/delegationtoken_test.go:62]: `t.Log(test.testRailID)`

File: ledger-core/virtual/integration/delegationtoken_test.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+				constructorPulse = server.GetPulse().PulseNumber
+
+				delegationToken payload.CallDelegationToken
+			)
+
+			delegationToken = server.DelegationToken(reference.NewRecordOf(class, outgoing.GetLocal()), server.GlobalCaller(), outgoing)
+			switch {
+			case test.fakeCaller:
+				delegationToken.Callee = server.RandomGlobalWithPulse()
```
Comment (ElSacramento) [inline ledger-core/virtual/integration/delegationtoken_test.go:112]: Caller

File: ledger-core/virtual/integration/delegationtoken_test.go (hunk 1 above)
Comment (MikhailZhukov) [reply ledger-core/virtual/integration/delegationtoken_test.go]: Fix

File: ledger-core/virtual/integration/delegationtoken_test.go (hunk 2 above)
Comment (MikhailZhukov) [reply ledger-core/virtual/integration/delegationtoken_test.go]: Fix

File: ledger-core/virtual/integration/delegationtoken_test.go (hunk 3 above)
Comment (MikhailZhukov) [reply ledger-core/virtual/integration/delegationtoken_test.go]: Fix

File: ledger-core/virtual/integration/delegationtoken_test.go [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+		t.Run(test.name, func(t *testing.T) {
+			t.Log("C5197")
+
+			mc := minimock.NewController(t)
+
+			server, ctx := utils.NewUninitializedServer(nil, t)
+			defer server.Stop()
+
+			runnerMock := logicless.NewServiceMock(ctx, mc, nil)
+			server.ReplaceRunner(runnerMock)
```
Comment (ElSacramento) [inline ledger-core/virtual/integration/delegationtoken_test.go:70]: I'm not sure, that you need to replace runner.

Comment (MikhailZhukov) [reply ledger-core/virtual/integration/delegationtoken_test.go]: fix

**Pull request merged**